### PR TITLE
breaking change: replaces unlawful instances with similar named funct…

### PR DESCRIPTION
**Pull request by CarstenKoenig**
Hi,

this implements the changes proposed in #4 - it removes the unlawful instances and replaces them with just the instance-functions.

Added benefit: can drop dependency on `control` ;)

### CODE REVIEW

File: src/Data/UndefinedOr.purs [hunk 1]
```diff
@@ -81,43 +90,43 @@ toUndefined = UndefinedOr
...
-
-instance plusUndefined :: Plus UndefinedOr where
-    empty = undefinedVal
+  eq ux@(UndefinedOr x) uy@(UndefinedOr y) = case unit of
+    _
+      | isUndefined ux && isUndefined uy -> true
+      | not (isUndefined ux) && not (isUndefined uy) -> eq x y
+    _ -> false
+
+-- | unlawful version of `map` - caution does not obey the functor laws
```
Comment (maurges) [inline src/Data/UndefinedOr.purs]: Nitpick: can you capitalize the comments and fix the punctuation? By fixing I mean `s/ - caution/. Caution:/`

File: src/Data/UndefinedOr.purs [hunk 2]
```diff
@@ -38,16 +38,27 @@
 -- | by clever use of typeclasses.
 -- |
 -- | You should especially Not use this over Maybe.
+-- |
+-- | Please also note that this (althoug it seems like a instance of) is not Functor
```
Comment (maurges) [inline src/Data/UndefinedOr.purs]: Style remark: remove brackets and change the sentence: `s/that this (.*) is not Functor/that this is not a Functor, \1 it/`
Also typo in 'although'

**Review by maurges — CHANGES_REQUESTED** [review]
Looks good with only some problems in comments

**Comment by maurges** [comment]
Oh and another thing: please bump the major version as this is a breaking change.

**Comment by CarstenKoenig** [comment]
Hi - I'll have a look once I get home - should be no problem
(sorry for the typos - I guess I should try and improve my English)

**Comment by CarstenKoenig** [comment]
Hi - I pushed the changes please have a look.

For the version you have to help me out a bit: I was under the impression that normally you'll use `pulp` (or I think `spago` can do as well) to bump up the version when you release the package - AFAIK that includes adding a Tag (`pulp` will do for you).

I don't actually see `1.0.1` mentioned anywhere - it's `1.0.0` in package.json - where do you want me to upgrade the version and what do you suggest? (SemVer should be `2.0.0` I guess)

File: src/Data/UndefinedOr.purs [hunk 3]
```diff
@@ -81,43 +91,43 @@ toUndefined = UndefinedOr
...
-
-instance plusUndefined :: Plus UndefinedOr where
-    empty = undefinedVal
+  eq ux@(UndefinedOr x) uy@(UndefinedOr y) = case unit of
+    _
+      | isUndefined ux && isUndefined uy -> true
+      | not (isUndefined ux) && not (isUndefined uy) -> eq x y
+    _ -> false
+
+-- | unlawful version of `map`. Caution does not obey the functor laws
```
Comment (maurges) [inline src/Data/UndefinedOr.purs]: Please capitalize it and put the colon after "Caution"

**Comment by maurges** [comment]
No problem for comments, in fact I might be hypercorrecting here, as English is not my native language as well. I just like it to look neat.

RE version: I don't even remember myself where it goes. But please do change what the package.json says at least, we'll figure out the rest later. You are correct, it should say 2.0

**Comment by CarstenKoenig** [comment]
Hi,

version in package.json is now at 2.0.0.

As far as I know for bower/purescript-package-set it's mainly done via Tags here. Psc-Package/Dhall will be fine with either commit/branch/tag but if you want to get this (re-)added to the current set it should have a current Tag I think.

**Review by maurges — CHANGES_REQUESTED** [review]
Just some more style changes and we're good to go (I forgor to press the "send" button yesterday..)

File: src/Data/UndefinedOr.purs (hunk 3 above)
Comment (CarstenKoenig) [reply src/Data/UndefinedOr.purs]: Ok - please check it now.

Comment (maurges) [reply src/Data/UndefinedOr.purs]: Thanks!

**Review by maurges — APPROVED** [review]

**Comment by maurges** [comment]
Great, now the only thing that's left is release it I guess. By the way, if the release process can be deferred to another person, I hereby grant you, @CarstenKoenig a permission to do a 2.0.0 release of this version of the code currently visible in the "master" branch for package sets, effective strictly between 2022-08-18 and 2022-09-01
=)

**Pull request merged**